res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -0,0 +1,2149 @@
+    if (!trackLoaded) {
+        return;
+    }
+
+    if (value && !cue) {
+        // play illumination:
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.playPauseDeck, true);
+        PioneerDDJSX.playFromCuePoint = true;
+        return;
+    }
+    if (!value && !cue) {
+        // pause illumination:
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.playPauseDeck, false);
+        PioneerDDJSX.playFromCuePoint = false;
+        return;
+    }
+    if (value && cue) {
+        // stop and go to cue point illumination:
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, true);
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.playPauseDeck, true);
+        PioneerDDJSX.cuePressed[deck] = false;
+        PioneerDDJSX.playFromCuePoint = true;
+    }
+    if (!value && cue) {
+        if (PioneerDDJSX.playFromCuePoint) {
+            // play from cue point illumination:
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, false);
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, true);
+            PioneerDDJSX.playFromCuePoint = false;
+        } else {
+            // set cue point illumination:
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, false);
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, true);
+            PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, false);
+        }
+        PioneerDDJSX.cuePressed[deck] = false;
+    }
+};
+
+PioneerDDJSX.playStutterIllumination = function(value, group, control) {
+    if (value) {
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, false);
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.cueDeck, true);
+    } else {
+        PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.playPauseDeck, true);
+    }
+};
+
+PioneerDDJSX.cueLed = function(value, group, control) {
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.cue, value);
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.shiftCue, value);
+};
+
+PioneerDDJSX.loadLed = function(value, group, control) {
+    var deck = PioneerDDJSX.channelGroups[group];
+    PioneerDDJSX.generalLedControl(PioneerDDJSX.nonPadLeds["loadDeck" + (deck + 1)], value);
+    PioneerDDJSX.illuminateFunctionControl(group, PioneerDDJSX.illuminationControl.loadedDeck, value);
+};
+
+PioneerDDJSX.reverseLed = function(value, group, control) {
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.censor, value);
+    PioneerDDJSX.nonPadLedControl(group, PioneerDD
```

Comment:
It IS possible, but then I have to rework a lot of code, since not only the LEDs act differently in "Serato Mode". All PAD functionalities have to be controlled/interpreted differently and the MIDI messages are not documented. The mode "DJ Software other than Serato" makes it a lot more easier to handle in a first step. I want to rework the mapping concerning the jog LEDs, "Serato Mode" and PAD functionalities in the future, would that be ok? I don't know how fast I can "explore" all the necessary MIDI messages and how much time it costs to rework the pad functionalities.